crates/yttrium/src/account_client.rs

```diff
@@ -131,8 +140,67 @@ impl AccountClient {
         .await
     }
 
-    pub async fn sign_message(&self, message: String) -> eyre::Result<String> {
-        todo!("Implement sign_message: {}", message)
+    pub fn prepare_sign_message(
+        &self,
+        message_hash: B256,
+    ) -> PreparedSignature {
+        if !self.safe {
+            unimplemented!(
+                "sign_message is not supported for non-safe accounts"
+            );
+        }
+
+        prepare_sign(
+            // TODO refactor class to parse Address on AccountClient
+            // initialization instead of lazily
+            self.owner.parse::<Address>().unwrap().into(),
+            U256::from(U64::from(self.chain_id)),
+            message_hash,
+        )
+    }
+
+    pub async fn do_sign_message(
+        &self,
+        signatures: Vec<OwnerSignature>,
+    ) -> eyre::Result<SignOutputEnum> {
+        if !self.safe {
+            unimplemented!(
+                "sign_message is not supported for non-safe accounts"
+            );
+        }
+
+        // TODO refactor class to create Provider on AccountClient
+        // initialization instead of lazily
+        let provider = ReqwestProvider::<Ethereum>::new_http(
+            self.config.endpoints.rpc.base_url.parse().unwrap(),
+        );
+
+        sign(
+            Owners {
+                owners: vec![self.owner.parse::<Address>().unwrap()],
+                threshold: 1,
+            },
+            self.get_address()
+                .await
+                .unwrap()
+                .parse::<Address>()
+                .unwrap()
+                .into(),
+            signatures,
+            &provider,
+            PaymasterClient::new(BundlerConfig::new(
+                self.config.endpoints.paymaster.base_url.parse().unwrap(),
+            )),
+        )
+        .await
+    }
+
+    pub async fn finalize_sign_message(
```

Comment:
Why do we need 3 steps to create a signature?

Reply:
Need 2 signatures unfortunately, 3rd step is to accept the second signature